authentication/src/main/java/io/camunda/authentication/entity/CamundaUser.java

```diff
@@ -8,25 +8,75 @@
 package io.camunda.authentication.entity;
 
 import java.util.Collections;
+import java.util.List;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.core.userdetails.User;
 
 public class CamundaUser extends User {
+
   private final Long userKey;
-  private final String name;
+  private final String displayName;
+
+  private boolean canLogout = true;
 
   public CamundaUser(
-      final Long userKey, final String name, final String username, final String password) {
+      final Long userKey, final String displayName, final String username, final String password) {
     super(username, password, Collections.emptyList());
     this.userKey = userKey;
-    this.name = name;
+    this.displayName = displayName;
+  }
+
+  public CamundaUser(final String displayName, final String username, final String password) {
+    super(username, password, Collections.emptyList());
+    userKey = null;
+    this.displayName = displayName;
+  }
+
+  public CamundaUser(
+      final String displayName,
+      final String username,
+      final String password,
+      final List<String> roles) {
+    super(username, password, roles.stream().map(SimpleGrantedAuthority::new).toList());
+    userKey = null;
```

Comment:
This constructor is used to create a `CamundaUser` from `UserEntity`. Why not setting the `userKey` from the `UserEntity.key` ?

Reply:
That would be nice if there was a key but this constructor is used in OperateUserDetailsService and SearchEngineUserDetailsService. and in none of them I can find the key in the Entity class. (UserEntity or TasklistEntity).

Reply:
oh my mistake! I thought is was the new UserEntity